pkg/api/controllers/profile.go

```diff
@@ -49,6 +48,40 @@ func (p *ProfilePortal) CreateProfile() {
 		return
 	}
 
+	//Validate StorageType is block or file
+	if profile.StorageType != "block" && profile.StorageType != "file" {
+		errMsg := fmt.Sprintf("parse profile request body failed : %v is invalid storagetype", profile.StorageType)
+		p.ErrorHandle(model.ErrorBadRequest, errMsg)
+		return
+	}
+
+	if profile.StorageType == "file" {
+		if pp := profile.ProvisioningProperties; !pp.IsEmpty() {
+			if ds := pp.DataStorage; !ds.IsEmpty() {
+				if len(ds.StorageAccessCapability) == 0 {
```

Comment:
should define supported access capabilities constants in hotpot, then check whether ds.StorageAccessCapability is contained in constants

Reply:
Validation added with enum, below is output. curl --header "Content-Type: application/json" \--data '{"name":"fileshare12", "description":"file share test", "storageType":"file", "provisioningProperties":{"DataStorage":{"StorageAccessCapability":["Read","Write","Execute", "Ex1"]}}, "ioConnectivity":{"accessProtocol":"NFS"}}}' http://127.0.0.1:50040/v1beta/e93b4c0934da416eb9c8d120c5d04d96/profiles 
{"code":400,"message":"Parse profile request body failed [Read Write Execute Ex1] is invalid storageaccesscapability"}